sklearn/inspection/_partial_dependence.py

```diff
@@ -258,44 +266,60 @@ def _partial_dependence_brute(
         is the number of instances in `X`, and `n_points` is the number of points
         in the `grid`.
     """
-    predictions = []
-    averaged_predictions = []
-
     if response_method == "auto":
         response_method = (
             "predict" if is_regressor(est) else ["predict_proba", "decision_function"]
         )
 
-    X_eval = X.copy()
-    for new_values in grid:
-        for i, variable in enumerate(features):
-            _safe_assign(X_eval, new_values[i], column_indexer=variable)
-
-        # Note: predictions is of shape
-        # (n_points,) for non-multioutput regressors
-        # (n_points, n_tasks) for multioutput regressors
-        # (n_points, 1) for the regressors in cross_decomposition (I think)
-        # (n_points, 2) for binary classification
-        # (n_points, n_classes) for multiclass classification
-        pred, _ = _get_response_values(est, X_eval, response_method=response_method)
+    n_samples, n_points = X.shape[0], len(grid)
+    max_memory_bytes = max_memory_mb * 1_048_576
```

Comment:
Here again, a megabyte is just one million bytes.  1_048_576 Bytes is a mebibyte which is a useless and non-standard way to count bytes:

https://en.wikipedia.org/wiki/Megabyte